caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    @Override
+    public void setUser(String userId) {
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.USER_ID_FIELD, userId);
+    }
+
+    @Override
+    public void setTenant(String tenantId) {
+        //  The tenant identifier is used as part of the Elasticsearch index name.
+        //  There are restrictions around the naming of the index including it must be lowercase
+        //  and not contain commas.
+        if(tenantId.contains(",")) {
+            //  Report invalid comma usage.
+            String errorMessage = "Invalid characters (i.e commas) in the tenant identifier: " + tenantId;
+            LOG.error(errorMessage);
+            throw new IllegalArgumentException(errorMessage);
+        }
+
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.TENANT_ID_FIELD, tenantId.toLowerCase());
+    }
+
+    @Override
+    public void setCorrelationId(String correlationId) {
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.CORRELATION_ID_FIELD, correlationId);
+    }
+
+    @Override
+    public void setEventType(String eventCategoryId, String eventTypeId) {
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.EVENT_CATEGORY_ID_FIELD, eventCategoryId);
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.EVENT_TYPE_ID_FIELD, eventTypeId);
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, String value) {
+        auditEventParams.add(new EventParam(name, "string", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, short value) {
+        auditEventParams.add(new EventParam(name, "short", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, int value) {
+        auditEventParams.add(new EventParam(name, "int", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, long value) {
+        auditEventParams.add(new EventParam(name, "long", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, float value) {
+        auditEventParams.add(new EventParam(name, "float", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, double value) {
+        auditEventParams.add(new EventParam(name, "double", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, boolean value) {
+        auditEventParams.add(new EventParam(name, "boolean", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, Date value) {
+        auditEventParams.add(new EventParam(name, "date", columnName, value));
+    }
+
+    @Override
+    public void send() throws Exception {
```

Comment:
I have added a small piece of javadoc here in my latest commit but I will revisit this in my new branch, that adds support for NO_PROXY, HTTP_PROXY and HTTPS_PROXY settings, as exceptions that are thrown by this method will likely change.